### PR TITLE
Feature/map frontend

### PR DESCRIPTION
**Pull request by omotoshoayomikun**

### CODE REVIEW

File: src/app/providers.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use client';
+import { ThemeProvider } from 'next-themes';
```
Comment (swoocn) [inline src/app/providers.tsx:2]: Is this working for you?  
I am getting a _Cannot find module 'next-themes' or its corresponding type declarations._. 🤔

File: .eslintrc.json [hunk 2]
```diff
@@ -1,9 +1,8 @@
 {
   "ignorePatterns": ["node_modules/", ".next/"],
-  "extends": [
-    "eslint:recommended",
-    "next/core-web-vitals",
-    "plugin:@typescript-eslint/recommended",
-    "plugin:prettier/recommended"
-  ]
+  "extends": "next",
```
Comment (swoocn) [inline .eslintrc.json:3]: Way to simplify! 👍

File: src/app/providers.tsx (hunk 1 above)
Comment (swoocn) [reply src/app/providers.tsx:2]: Below is a screenshot of the compile-time error and the console output when I attempt to run the app locally.

image

File: src/components/Models/LangModel/LangModel.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    },
+  ];
+
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const { setTheme, theme } = useTheme();
+
+  const handleLanguageChange = (code: string) => {
+    const slipPathname = pathname.split('/').slice(2);
```
Comment (swoocn) [inline src/components/Models/LangModel/LangModel.tsx:73]: Can `pathname` ever be null or undefined?  Would it be worth putting a null check here?

Comment (swoocn) [reply src/components/Models/LangModel/LangModel.tsx:73]: It appears `npm run build` is failing possibly for this reason.

**Comment by swoocn** [comment]
Overall, great attempt with the implementation for Task #1 @omotoshoayomikun!  I went ahead and made some minor structural adjustments as well as left you some PR comments.  

Btw, were you able to get the app to run locally via `npm run dev`?  Asking because I am getting a bunch of DOMException runtime errors. 🤔

**Review by soleil00 — APPROVED** [review]

**Review by swoocn — APPROVED** [review]

**Pull request merged**